src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -346,46 +448,105 @@ private AIState cookFood()
         {
+            // TODO: why do we care if the furnace is not burning?  burning && food to cook, maybe, but not burning in general.
+            // TODO: Why do we care about food in the cook slot but not fuel? both are required.
+            // TODO: It makes more sense to find ovens which have cooked food (no matter what else is going on), or will have cooked food (either because it is in the process of
+            // being cooked or because there is uncooked food plus fuel)
             final TileEntity entity = world.getTileEntity(pos);
             if (entity instanceof TileEntityFurnace && !((TileEntityFurnace) entity).isBurning()
                     && (!ItemStackUtils.isEmpty(((TileEntityFurnace) entity)
                     .getStackInSlot(RESULT_SLOT))
                     || !ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(COOK_SLOT))))
             {
-                worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.retrieving"));
                 return pos;
             }
         }
         return null;
     }
 
+    /**
+     * Specify that we dump inventory after every action.
+     * @see com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic#getActionsDoneUntilDumping()
+     * @return 1 to indicate that we dump inventory after every action
+     */
     @Override
     protected int getActionsDoneUntilDumping()
     {
         return 1;
     }
 
+    /**
+     * Execute tasks in this order: Retrieve food or cook food in furnaces, request food, serve food, cook food, other tasks.
+     *
+     * If any furnace has food waiting to be cooked, Find the first furnace in the building which is not cooking and either has a cooked result or has an uncooked input
+     * and walk to that furnace's position by transitioning to COOK_RETRIEVE_FOOD.
+     * If any furnace has food waiting to be cooked, and if no such furnace, then COOK_COOK_FOOD.
+     * If no food in the building nor inventory, set status to "Waiting For Food" and request food if we haven't already done so. In any case, repeat this state.
+     * If we have food in building or inventory, find all citizens nearby that are not cooks and have a zero or negative saturation.
+     * If there is a hungry citizen and we have food in inventory, then COOK_SERVE
+     * If there is a hungry citizen and we have no food in inventory, then COOK_GATHER_FOOD_FROM_BUILDING
+     * if no hungry citizens and we have less than 64 * building-level food stores, request food if we haven't already done so.
+     * If there is uncooked food in building or inventory, then COOK_COOK_FOOD.
+     * If any furnace is idle and has cooked food, then COOK_RETRIEVE_FOOD.
+     * If any furnace is idle and has uncooked food, then COOK_COOK_FOOD.
+     * Otherwise, set status to "Idling", then delay and START_WORKING
+     *
+     * @return next AIState
+     */
     private AIState startWorking()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 8, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "startWorking" is 17 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)